### DISCUSSION TITLE
Notification not showing when push is received

### DISCUSSION THREAD

**Discussion by jesse-mta** [shinyorg/shiny#1417] [Q&A]
Hi, 

Quick question. I'm using Shiny.Push in my .NET MAUI app. I've got everything wired up and I can see the notification being received in OnReceived method. But the notification is not displaying (native notification popup). Am I to implement this functionality in my app or does shiny take care of that. I'm I supposed to use Shiny.Notification to display the native notification  popup.

Jus to be clear I'm talking about this:

![image](https://github.com/shinyorg/shiny/assets/159189842/163c245c-748d-4d69-ab2e-a3f950a0b1e1)

**Comment by aritchie** [comment]
The platform itself will take care of it if you include a notification payload

**Reply by jesse-mta** [reply]
Ok, I see the notification when the app is not in foreground. When app is in foreground I don't see the notification. I guess I have to handle that in my app when app is in foreground?

**Reply by munkii** [reply]
For non-silent Push Notifications on Android we only see OnReceived fire when the app is in the Foreground. Using this knowledge we know that the Notifcation will not appear so we take the data from the PushNotifcation and create a local Notification to be displayed.

I'm sure @aritchie will jump in if that is madness but that is what we do.

**Reply by jesse-mta** [reply]
Thanks. Great library btw.

**Reply by munkii** [reply]
I must stress "we" in the context of my comment is the app I am writing and as a consumer of Shiny. That is what we do in our app not what Shiny does. I can take no credit for Shiny. :-)

**Reply by jesse-mta** [reply]
@aritchie , please clarify what "include a notification payload". Is this what you mean by "payload".  I am sending a notification like so. I'm I missing something? Does it matter if app is in foreground or not?

{
  "notification": {
    "title": "New Message",
    "body": "You've received a new message!",    
    "click_action": "SHINY_PUSH_NOTIFICATION_CLICK"
  }
}

**Reply by aritchie** [reply]
Your answers are here: https://github.com/shinyorg/pushtester/tree/main
